src/CopyrighFixer/CopyrighFixer/signer.cpp

```diff
@@ -19,44 +20,90 @@ bool Signer::checkSign(const Config &objConf) {
     Config currConfig = objConf;
 
     QDir currentFolder(currConfig.getSrcDir());
-    currentFolder.setFilter(QDir::Dirs | QDir::Files);
+    currentFolder.setFilter(QDir::Dirs | QDir::Files | QDir::NoDotAndDotDot);
 
     QFileInfoList folderItems(currentFolder.entryInfoList());
 
     for (auto i_file: folderItems) {
 
-        if (i_file.fileName() == "." || i_file.fileName() == "..") {
-            continue;
-        }
+        if (i_file.isDir()) {
+
+            currConfig.setSourceDir(i_file.filePath());
+            checkSign(currConfig);
 
-        if (!i_file.isDir()) {
+        } else {
 
             IFileManager *currFM = searchFileByExt(i_file.suffix());
-            Signature objSing = currConfig.getSignVal();
+            Signature signFromFile;
+            Signature signForWrite;
 
             if (currFM == nullptr) {
                 return false;
             }
 
-            bool valRead = currFM->read(i_file.filePath(), objSing);
-            if (!valRead) {
+            if (!currFM->read(i_file.filePath(), signFromFile)) {
                 return false;
             }
 
-            currConfig.setSingValue(objSing);
+            signForWrite = mergeSign(currConfig.getSignVal(), signFromFile);
 
-            bool valWrite = currFM->write(i_file.filePath(), objSing);
-            if (!valWrite) {
+            if (!currFM->write(i_file.filePath(), signForWrite)) {
                 return false;
             }
+        }
+    }
+
+    return true;
+}
+
+const Signature Signer::mergeSign(const Signature &userSign, const Signature &fileSign) const {
+
+    if (!fileSign.isValid()) {
+        return userSign;
+    }
+
+    if (userSign.getLicenseTitle() == fileSign.getLicenseTitle()) {
+
+        if (!fileSign.isValid()) {
+            return userSign;
+        }
+
+        if (userSign.getMapOwn().size() > 1) {
+            QuasarAppUtils::Params::log("Config signature contains more owners.",
+                                        QuasarAppUtils::VerboseLvl::Warning);
+            return fileSign;
+        }
+
+        Signature signForSing = fileSign;
+        if (userSign.getMapOwn().cbegin().value().getOwnerName() == signForSing.getMapOwn().cbegin().value().getOwnerName()) {
+
+            QMap<int, CopyrighFixer::Owner> mapOwners = signForSing.getMapOwn();
+            mapOwners.remove(fileSign.getMapOwn().cbegin().key());
+            mapOwners.insert(userSign.getMapOwn().cbegin().key(),
+                             userSign.getMapOwn().cbegin().value());
+            signForSing.setMapOwners(mapOwners);
+
+            return signForSing;
```

Comment:
This function is hard for readable and not scalable.
I recommends use code without else operators. 

example:
```cpp
if (title != title ) {
   case1();
   return;
}

if (licenseText != licenseText) {
    case2();
    return;
}

if (licenseOwner != licenseOwner){
    case3;
    return;
}

default case; // update time
```

where a caseN is a separated methods with processing a separated case.